### PR TITLE
fix duplicate testing when unique suite name

### PR DESCRIPTION
**Pull request by vjeffrey**
fixes https://github.com/chef/kitchen-inspec/issues/109

trying to work on a test for this....

### CODE REVIEW

File: lib/kitchen/verifier/inspec.rb [hunk 1]
```diff
@@ -125,10 +125,27 @@ def local_suite_files
...
       # @return [Array<String>] array of suite directories or remote urls
       # @api private
       def collect_tests
+        puts 'local suite files', local_suite_files
+        puts 'config inspec tests', config[:inspec_tests]
+        # if check for duplicates returns true, only send back the value for config[:inspec_tests]
+        return config[:inspec_tests] if check_for_duplicates(local_suite_files, config[:inspec_tests])
+
         # get local tests and get run list of profiles
         (local_suite_files + config[:inspec_tests]).compact
```
Comment (chris-rock) [inline lib/kitchen/verifier/inspec.rb]: Maybe we should simply use `(local_suite_files + config[:inspec_tests]).compact.uniq`? We also need to double-check that this works with the new complex objects

Comment (vjeffrey) [inline lib/kitchen/verifier/inspec.rb]: local suite files comes in with the full path:
```
** local suite files
/Users/vjeffrey/test-kitchen-inspec/test/recipes/master
## config[:inspec_tests]
{:path=>"test/recipes/master"}
```

Comment (vjeffrey) [inline lib/kitchen/verifier/inspec.rb]: oh, shoot, the complex objects isn't working well. darnit.

Comment (chris-rock) [inline lib/kitchen/verifier/inspec.rb]: Okay, maybe we do the following:
1.) we rewrite local suite files to return a complex object like 
```
{:path=>"/Users/vjeffrey/test-kitchen-inspec/test/recipes/master"}
```
2.) We also resolve the `inspec_tests` to absolute paths before we hand them over to inspec
3.) Extend your `check_for_duplicates` to only work with objects?

Comment (vjeffrey) [inline lib/kitchen/verifier/inspec.rb]: :+1:

**Comment by vjeffrey** [comment]
k, so as part of this change, since we are harmonizing profile targets, there's an error that is raised if the inspec tests that are specified are not specified in the defined format (like `path: path/to/file`  or `url: url`).  i was initially trying to support backwards compatibility, but it was getting complicated with transforming the hashes....so i went this route.  if that's too much, we can work around to support non-hashes....but it seems like since we're pushing this hash change across audit cookbook and here, it would make sense to enforce it.  The error message is nice and verbose so the user should understand pretty quickly...

**Comment by chris-rock** [comment]
@vj I think we stay backwards compatible for now (easier upgrade for users), but we do duplicate checking only for the new syntax.

**Comment by chris-rock** [comment]
Nice work @vjeffrey

**Pull request merged**